code/modules/mob/living/carbon/xenomorph/mutators/strains/hivelord/resin_whisperer.dm

```diff
@@ -67,6 +68,11 @@
 	action_type = XENO_ACTION_CLICK
 
 /datum/action/xeno_action/activable/secrete_resin/remote/use_ability(atom/target_atom, mods)
+	var/mob/living/carbon/xenomorph/hivelord/hivelord_mob = owner
+	if(!hivelord_mob.on_weeds())
```

Comment:
![image](https://user-images.githubusercontent.com/4447185/233054144-edfcf37c-b73a-4713-8a06-e75aeea74d85.png)
This ability is also being used by the queen, so trying to treat her as a hivelord breaks her ability to build from ovipositor completely.

Reply:
> This ability is also being used by the queen, so trying to treat her as a hivelord breaks her ability to build from ovipositor completely.

My bad I didn't realize the queen used the same ability.
I went ahead and added a check for it.